### ISSUE TITLE
[EmailField] setPattern should not override the default email format but make it stricter

### ISSUE THREAD

**Issue by vursen** [vaadin/flow-components#6561]
### Describe your motivation

Using `setPattern` currently overrides the default RFC 5322 email format entirely. Furthermore, setting `setPattern(null)` does not restore the default regexp but removes it completely, turning EmailField into a regular text field. 

In other words, if you want to enforce a specific domain, for example, you have to take the standard email regexp,  modify it, and set it back using `setPattern`. This process is inconvenient and prone to mistakes, which can accidentally allow invalid emails to pass.

### Describe the solution you'd like

This behavior is arguably incorrect. Instead, setPattern should add additional criteria on top of the default format, making it stricter. Similarly, setting `setPattern(null)` should simply remove the additional criteria while preserving the default format.

In other words, if you want to enforce a specific domain, for example, you should only need to provide a regular expression that will ensure that the email ends with the desired domain, e.g: `/.+@vaadin\.com$/`

**Comment by vursen** [comment]
The proposal behavior aligns with that of the native `<input type="email">`:

> Since inputs of type email validate against both the standard email address validation and the specified [pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#pattern), you can implement this easily.
>
> https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email#pattern_validation
